vermouth/rcsu/contact_map.py

```diff
@@ -12,10 +12,653 @@
+
+
+def calculate_contacts(vdw_list, atypes, coords, res_serial, nresidues):
+    """
+    run the contact calculation functions
+
+    vdw_list: np.array
+        list of the vdw radii of the atoms in the system
+    atypes: np.array
+        list of the atom types in the system to determine the nature of contacts
+    coords: nx3 array
+        coordinates of all the atoms in the system
+    res_serial: np.array
+        list of the serial residue number of each atom in the system
+    nresidues: int
+        number of residues in the system
+    """
+
+    # some initial definitions of variables that we need
+    fib = 14
+    fiba, fibb = 0, 1
+    for _ in range(fib):
+        fiba, fibb = fibb, fiba + fibb
+
+    natoms = len(coords)
+
+    # all_vdw = np.array([x for xs in
+    #                     [[PROTEIN_MAP[i][j]['vrad'] for j in PROTEIN_MAP[i].keys()]
+    #                      for i in PROTEIN_MAP.keys()]
+    #                     for x in xs])
+    vdw_max = 1.88  # all_vdw.max()
+
+    # make the KDTree of the input coordinates
+    coords_tree = KDTree(coords)
+
+    # calculate the OV contacts of the molecule
+    over = calculate_overlap(coords_tree, vdw_list, natoms, vdw_max)
+
+    # Calculate the CSU contacts of the molecule
+    hit_results = calculate_csu(coords, vdw_list, fiba, fibb, natoms, coords_tree, vdw_max)
+
+    # find the types of contacts we have
+    contactcounter_1, stabilisercounter_1, destabilisercounter_1 = contact_types(hit_results, natoms, atypes)
+
+    # transform the resolution between atoms and residues
+    overlapcounter_2 = atom2res(over, res_serial, nresidues, norm=True)
+    contactcounter_2 = atom2res(contactcounter_1, res_serial, nresidues)
+    stabilisercounter_2 = atom2res(stabilisercounter_1, res_serial, nresidues)
+    destabilisercounter_2 = atom2res(destabilisercounter_1, res_serial, nresidues)
+
+    return overlapcounter_2, contactcounter_2, stabilisercounter_2, destabilisercounter_2
+
+
+def get_contacts(nresidues, overlaps, contacts, stabilisers, destabilisers, nodes, G):
+    contacts_list = []
+    for i1 in range(nresidues):
+        for i2 in range(nresidues):
+            over = overlaps[i1, i2]
+            cont = contacts[i1, i2]
+            stab = stabilisers[i1, i2]
+            dest = destabilisers[i1, i2]
+            rcsu = 1 if (stab - dest) > 0 else 0
+
+            if (over > 0 or cont > 0) and (i1 != i2):
+                a = np.where(nodes == i1)[0][0]
+                b = np.where(nodes == i2)[0][0]
+                if over == 1 or (over == 0 and rcsu == 1):
+                    # this is a OV or rCSU contact we take it
+                    contacts_list.append((int(G.nodes[a]['resid']), G.nodes[a]['chain'],
+                                          int(G.nodes[b]['resid']), G.nodes[b]['chain']))
+
+    return contacts_list
+
+
+def write_contacts(nresidues, overlaps, contacts, stabilisers, destabilisers, nodes, ca_pos, G):
```

Comment:
This function repeats a lot of code from get_contacts. Can't you use the contacts list that generates as input here?